include/pybind11/detail/type_caster_base.h

```diff
@@ -511,16 +501,27 @@ PYBIND11_NOINLINE std::string error_string() {
             Py_INCREF(f_code);
 #    endif
             int lineno = PyFrame_GetLineNumber(frame);
-            errorString += "  " + handle(f_code->co_filename).cast<std::string>() + "("
-                           + std::to_string(lineno)
-                           + "): " + handle(f_code->co_name).cast<std::string>() + "\n";
+            result += std::string("  ") + handle(f_code->co_filename).cast<std::string>() + '('
+                      + std::to_string(lineno)
+                      + "): " + handle(f_code->co_name).cast<std::string>() + '\n';
             frame = frame->f_back;
             Py_DECREF(f_code);
         }
-    }
 #endif
```

Comment:
Might be nice to add `// !defined(PYPY_VERSION)` after the existing `#endif`, while we're at it.